### ISSUE TITLE
better error message for to_netcdf -> unlimited_dims

### ISSUE THREAD

**Issue by mathause** [pydata/xarray#2036]
#### Code Sample, a copy-pastable example if possible

```python
# Your code here
import numpy as np
import xarray as xr
x = np.arange(10)
da = xr.Dataset(data_vars=dict(data=('dim1', x)), 
               coords=dict(dim1=('dim1', x), dim2=('dim2', x)))
da.to_netcdf('tst.nc', format='NETCDF4_CLASSIC', unlimited_dims='dim1')

```
#### Problem description

This creates the error `RuntimeError: NetCDF: NC_UNLIMITED size already in use`. With `format='NETCDF4'` silently creates the dimensions `d`, `i`, `m`, and `\1`.

The correct syntax is `unlimited_dims=['dim1']`. 

With `format='NETCDF4_CLASSIC'` and  `unlimited_dims=['dim1', 'dim2']`, still raises the not-so-helpful `NC_UNLIMITED` error.

I only tested with netCDF4 as backend.

#### Expected Output

* better error message
*  work with `unlimited_dims='dim1'`

#### Output of ``xr.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.5.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.120-45-default
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_GB.UTF-8
LOCALE: en_GB.UTF-8

xarray: 0.10.2
pandas: 0.22.0
numpy: 1.14.2
scipy: 1.0.1
netCDF4: 1.3.1
h5netcdf: 0.5.0
h5py: 2.7.1
Nio: None
zarr: None
bottleneck: 1.2.1
cyordereddict: 1.0.0
dask: 0.17.2
distributed: 1.21.5
matplotlib: 2.2.2
cartopy: 0.16.0
seaborn: 0.8.1
setuptools: 39.0.1
pip: 9.0.3
conda: None
pytest: 3.5.0
IPython: 6.3.0
sphinx: 1.7.2
</details>

**Comment by mathause** [comment]
This could be relevant here:
https://github.com/pydata/xarray/pull/1170/files

**Comment by mathause** [comment]
https://github.com/shoyer/h5netcdf/issues/42

**Comment by jhamman** [comment]
closed via #2154
